### DISCUSSION TITLE
[knitr] Cache: Examples for the cache feature | /knitr/demo/cache/

### DISCUSSION THREAD

**Discussion by yihui** [rbind/yihui#778] [Comments]
https://yihui.org/knitr/demo/cache/

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Unfortunately, no. The reason is that dependencies are built at the parsing time of a parent document, when the child documents are not parsed yet. However, there is at least one workaround. For example, you can use an arbitrary chunk option in the child chunk like foo=x where x comes from mainchunk, so that whenever x is changed, this child chunk will be updated accordingly.

> Originally posted on 2012-08-18 05:14:04

**Reply by yihui** [reply]
The 0.7 behavior is expected; 0.4 is dangerous because it may bring potential chunk conflicts of which users can be unaware. For example, the noname chunk in the child doc can override the chunk with the same label in the parent doc. In 0.7 I required chunk labels to be unique across all the documents to avoid this problem.

> Originally posted on 2012-08-19 01:16:47

**Reply by yihui** [reply]
@google-9efc2d0be4fe563faf44e811749aa874:disqus I do not quite understand what you mean. Perhaps example 20 at <https://github.com/yihui/knitr-examples> ?

> Originally posted on 2012-10-26 03:59:24

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
You should see this warning the first time you compile the document, but I think it should go away the next time.

> Originally posted on 2012-12-14 22:51:54

**Reply by yihui** [reply]
Did you cache the chunk containing dep_auto() as well? You are not supposed to cache that one.

> Originally posted on 2012-12-28 01:33:38

**Reply by yihui** [reply]
I still cannot reproduce the problem. It works fine for me. Are you using the latest version of knitr?

> Originally posted on 2012-12-28 16:40:05

**Reply by yihui** [reply]
Sorry I just figured out the bug yesterday, and it has been fixed in the devel version: <https://github.com/yihui/knitr/commit/a35827cc>

Please install the devel version on Github and see if the bug has gone. Thanks!

library(devtools); install_github('knitr', 'yihui')

> Originally posted on 2013-01-08 22:46:42

**Reply by yihui** [reply]
Note that the new version (1.0) has been released to CRAN now, so it should work if you install from CRAN directly.

> Originally posted on 2013-01-15 01:20:15

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
See issue 457 <https://github.com/yihui/knitr/issues/457> and also the section "Even more stuff for cache?" in this page.

> Originally posted on 2013-01-15 03:38:39

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Hi, cache=FALSE is the correct way of turning off cache. Once cache is turned off, your code will be re-evaluated every time. I do not understand your question, because it is certainly expected that sample(1:10, 3) produces different results every time.

> Originally posted on 2013-10-21 04:02:45

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
This sounds indeed weird. Sorry, but I cannot do anything without a minimal reproducible example.

> Originally posted on 2014-07-06 16:18:56

[a comment by an automated account is not shown]

**Comment by przmv** [comment]
Thank you very much for your awesome work on this package!

Does caching feature also work with language engines other than R?

Thanks,
Petr

**Reply by yihui** [reply]
Yes, it does.
